### PR TITLE
Added commit status fetching functionality

### PR DESCRIPTION
**Pull request by Benjamin-Dobell**
Basically I needed this functionality (and a few other things I'll be adding), but figured I might as well try do it "properly". Let me know what needs changing :smile:

### CODE REVIEW

**Comment by Benjamin-Dobell** [comment]
Hmmm, a Swift spec is failing. As far as I can tell that must have been an existing issue.

```
/Users/travis/build/octokit/octokit.objc/OctoKitTests/SwiftSpec.swift:11:8: error: module file's minimum deployment target is ios9.0 v9.0: /Users/travis/Library/Developer/Xcode/DerivedData/OctoKit-cjajgpsfwpostubkfrhudkncaxub/Build/Products/Debug-iphonesimulator/Quick.framework/Modules/Quick.swiftmodule/i386.swiftmodule
import Quick
       ^
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━���━━━━
      ✗ Compile Swift source files (2466 ms)
```

File: OctoKit/OCTCommitCombinedStatus.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+#import "OCTObject.h"
+
+@class OCTRepository;
+
+// A combined status for a commit.
+@interface OCTCommitCombinedStatus : OCTObject
+
+// The combined state of the commit; one of "failure", "pending" or "success".
+@property (nonatomic, copy, readonly) NSString *state;
```
Comment (joshaber) [inline OctoKit/OCTCommitCombinedStatus.h]: What do you think about making this an enum instead of a string?

File: OctoKit/OCTCommitStatus.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+// The date at which the status was originally created.
+@property (nonatomic, copy, readonly) NSDate *creationDate;
+
+// The date the status was last updated. This will be equal to
+// creationDate if the status has not been updated.
+@property (nonatomic, copy, readonly) NSDate *updatedDate;
+
+// The state of the commit; one of "failure", "pending" or "success".
+@property (nonatomic, copy, readonly) NSString *state;
```
Comment (joshaber) [inline OctoKit/OCTCommitStatus.h]: (Same as above about enum vs. string.)

**Comment by joshaber** [comment]
Thanks @Benjamin-Dobell! Looks like CI is passing now. Just one question.

**Comment by Benjamin-Dobell** [comment]
Regarding the points above suggesting the use of enums over strings; we definitely want some string constants, but should avoid the use of enums. I was actually going to bring-up the problematic use of enums in another pull request that I'll be submitting *very* shortly (webhook support).

**Comment by joshaber** [comment]
Good points in the other PR :+1: String constants sound good.

**Comment by Benjamin-Dobell** [comment]
> Good points in the other PR :+1: String constants sound good.

Thanks, not all those points apply here - because we wouldn't be using a bitfield here, just a regular enum. Nonetheless, it does still have limitations if a new state is introduced: we'd either choke during transformation on the unknown state, or present inaccurate information to API consumers.

I'll definitely add some string constants. Thanks for your time/feedback!

**Comment by shenAlexy** [comment]
Good points

**Comment by joshaber** [comment]
Sorry for the delay @Benjamin-Dobell! I didn't realize the PR had been updated. Thanks for your work!

**Pull request merged**